src/libsyntax/feature_gate.rs

```diff
@@ -820,7 +826,7 @@ impl<'a> Context<'a> {
             // feature gate checking. Macro gating runs
             // before the plugin attributes are registered
             // so we skip this then
-            if !is_macro {
+            if !is_macro && !attr::is_used(attr) {
```

Comment:
Using mark_used appears to only disable the unused_attribute lint, not the custom_attributes error. So without this change the compiler will still give the custom attribute error for used attributes. The only alternative I can see is registering the attribute name in plugin_attributes.